code/game/atoms_movable.dm

```diff
@@ -1211,7 +1211,8 @@
  */
 /atom/movable/proc/on_changed_z_level(turf/old_turf, turf/new_turf, same_z_layer, notify_contents = TRUE)
 	SHOULD_CALL_PARENT(TRUE)
-	SEND_SIGNAL(src, COMSIG_MOVABLE_Z_CHANGED, old_turf, new_turf, same_z_layer)
+	if(SEND_SIGNAL(src, COMSIG_MOVABLE_Z_CHANGED, old_turf, new_turf, same_z_layer) & Z_CHANGE_PREVENTED)
+		return
```

Comment:
This does not seem to be the appropriate place to put this

You are not preventing the z change, this proc is "the z change has already happened and we're handing the consequences" - all you're doing is preventing said consequences (updating visuals, primarily)